cli/command_snapshot_migrate.go

```diff
@@ -96,18 +107,89 @@ func runMigrateCommand(ctx context.Context, destRepo *repo.Repository) error {
 	}
 
 	wg.Wait()
+	progress.FinishShared()
+	printStderr("\r\nMigration finished.\n")
 
 	return nil
 }
 
+func openSourceRepo(ctx context.Context) (*repo.Repository, error) {
+	pass, ok := repo.GetPersistedPassword(ctx, *migrateSourceConfig)
+	if !ok {
+		var err error
+
+		if pass, err = getPasswordFromFlags(ctx, false, false); err != nil {
+			return nil, errors.Wrap(err, "source repository password")
+		}
+	}
+
+	sourceRepo, err := repo.Open(ctx, *migrateSourceConfig, pass, applyOptionsFromFlags(ctx, nil))
+	if err != nil {
+		return nil, errors.Wrap(err, "can't open source repository")
+	}
+
+	return sourceRepo, nil
+}
+
+func migratePoliciesForSources(ctx context.Context, sourceRepo, destRepo *repo.Repository, sources []snapshot.SourceInfo) error {
+	for _, si := range sources {
+		if err := migrateSinglePolicy(ctx, sourceRepo, destRepo, si); err != nil {
+			return errors.Wrapf(err, "unable to migrate policy for %v", si)
+		}
+	}
+
+	return nil
+}
+
+func migrateAllPolicies(ctx context.Context, sourceRepo, destRepo *repo.Repository) error {
+	policies, err := policy.ListPolicies(ctx, sourceRepo)
+	if err != nil {
+		return errors.Wrap(err, "unable to list source policies")
+	}
+
+	for _, pol := range policies {
+		if err := migrateSinglePolicy(ctx, sourceRepo, destRepo, pol.Target()); err != nil {
+			log(ctx).Warningf("unable to migrate policy for %v: %v", pol.Target(), err)
```

Comment:
It is surprising that the function signature has an error return value, but then it swallows the errors in some cases. it'd may be desirable to not swallow the error. Thoughts?